src/SymfonyCache/CacheEvent.php

```diff
@@ -39,10 +38,10 @@ class CacheEvent extends Event
     private $response;
 
     /**
-     * @param HttpCache $kernel  The kernel raising with this event.
-     * @param Request   $request The request being processed.
+     * @param CacheManipulationInterface $kernel  The kernel raising with this event.
+     * @param Request                    $request The request being processed.
      */
-    public function __construct(HttpCache $kernel, Request $request)
+    public function __construct(CacheManipulationInterface $kernel, Request $request)
```

Comment:
i could remove the typehint and do an instanceof check in the constructor for either HttpCache *or* the interface.